libs/StatusGoQt/src/StatusGo/Chat/ChatDto.cpp

```diff
@@ -0,0 +1,241 @@
+    j = {{"id", d.id},
+         {"name", d.name},
+         {"description", d.description},
+         {"color", d.color},
+         {"emoji", d.emoji},
+         {"active", d.active},
+         {"timestamp", d.timestamp},
+         {"lastClockValue", d.lastClockValue},
+         {"deletedAtClockValue", d.deletedAtClockValue},
+         {"readMessagesAtClockValue", d.readMessagesAtClockValue},
+         {"unviewedMessagesCount", d.unviewedMessagesCount},
+         {"unviewedMentionsCount", d.unviewedMentionsCount},
+         {"canPost", d.canPost},
+         {"alias", d.alias},
+         {"identicon", d.icon},
+         {"muted", d.muted},
+         {"position", d.position},
+         {"communityId", d.communityId},
+         {"profile", d.profile},
+         {"joined", d.joined},
+         {"syncedTo", d.syncedTo},
+         {"syncedFrom", d.syncedFrom},
+         {"highlight", d.highlight},
+         {"categoryId", d.categoryId},
+         {"permissions", d.permissions},
+         {"chatType", d.chatType},
+         {"members", d.members},
+        };
+}
+
+void Chats::from_json(const json& j, ChatDto& d) {
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(id)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(name)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(description)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(color)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(emoji)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(active)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(timestamp)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(lastClockValue)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(deletedAtClockValue)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(readMessagesAtClockValue)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(unviewedMessagesCount)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(unviewedMentionsCount)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(canPost)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(alias, "alias", false)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(icon, "icon", false)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(muted)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(position, "position", false)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(communityId)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(profile, "profile", false)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(joined)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(syncedTo)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(syncedFrom)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(highlight, "highlight", false)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(permissions, "permissions", false)
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(chatType)
+
+    STATUS_READ_NLOHMAN_JSON_PROPERTY(categoryId, "categoryId", false)
+    if (!j.contains("categoryId")) {
+        // Communities have `categoryID` and chats have `categoryId`
+        // This should be fixed in status-go, but would be a breaking change
+        STATUS_READ_NLOHMAN_JSON_PROPERTY(categoryId, "categoryID", false)
+    }
+
+    // Add community ID if needed
+    if (!d.communityId.isEmpty() && !d.id.contains(d.communityId)) {
+        d.id = d.communityId + d.id;
```

Comment:
This looks like logic, why there is logic in the parsing of a DTO?

Reply:
@stefandunca that's a long story... a consequence of the transition from chats to channel groups. The same chat dto is used for a few status go responses which are not consistent.